### ISSUE TITLE
[a11y] avoid TYPE_WINDOW_STATE_CHANGED to announce tooltips

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#86577]
See https://github.com/flutter/engine/blob/eaf70ca01903d25f2cd7aad2f0f9ce754e1e0e62/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L268-L273.

As of Android 11 newer versions of TalkBack may not announce the message provided to the TYPE_WINDOW_STATE_CHANGED event. The alternative way for announcing tooltips is TBD. We may have to look at the Android source to see how the a11y for the native tooltip works or reverse-engineer the right way from TalkBack's debug logs when activating a native tooltip.

**Comment by goderbauer** [comment]
I wonder if this new-ish a11y API could also be helpful?

https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setTooltipText(java.lang.CharSequence)

**Comment by goderbauer** [comment]
Related: https://github.com/flutter/flutter/issues/86575

**Comment by goderbauer** [comment]
(Might also be worthwhile to check weather native tooltips still get announced at all on the latest Android/TalkBack version)

**Comment by goderbauer** [comment]
I got some more information on this:

* Tooltips should not dispatch a TYPE_WINDOW_STATE_CHANGED event
* TalkBack is not expected to read out tooltips when they show up on screen
* [setTooltipText](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setTooltipText(java.lang.CharSequence)) is to be used to tell a11y systems about Tooltips.

**Comment by blasten** [comment]
Thanks @goderbauer. 

> Tooltips should not dispatch a TYPE_WINDOW_STATE_CHANGED event

do you happen to know if this is a new behavior or is this also true prior to S?

**Comment by blasten** [comment]
I searched a bit. It appears that this is a workaround that offered backward compatibility below < 26. Since tooltips were introduced in 26. 

28 added `setTooltipText`, which is what we should be using.

**Comment by chunhtai** [comment]
The tooltip pop up is no longer announced in latest android, do we still need to use setTooltipText? It is also not clear to me when we should setToolTip for the accessibility node info, we can't attach the tooltip to the a11ynodeinfo unless the system ask for the specific node.

**Comment by chunhtai** [comment]
Ok so the tooltip will get append to the end of focus announcement when the node receive the a11y focus. Seems like we should do the same

**Comment by chunhtai** [comment]
tooltip repro

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Testing nested talkback app',
      theme: ThemeData(
        primaryColor: Colors.white,
      ),
      home: HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Testing'),
      ),
      body: Tooltip(message: 'my tool tip', child: Text('text'),),
    );
  }
}

```

**Comment by chunhtai** [comment]
The last pr is abandoned due to I am busy with other high prioirty issue
https://github.com/flutter/flutter/pull/87684
will get back to it once I have time
